validator/sawtooth_validator/journal/chain.py

```diff
@@ -473,3 +480,27 @@ def on_block_received(self, block):
         # pylint: disable=broad-except
         except Exception as exc:
             LOGGER.exception(exc)
+
+    def _set_chain_head(self, block):
+        # This is used by journal when it has recieved the genesis block from
+        # the genesis validator
+        if block.previous_block_id == NULL_BLOCK_IDENTIFIER:
+            self._block_store.set_chain_head(block.identifier)
+
+            self._block_store[block.identifier] = block
+            self._chain_head = block
+            self._notify_on_chain_updated(self._chain_head)
+
+            self._save_block_chain_id(block.identifier)
+        else:
+            LOGGER.warning("Cannot set intial chain head, this is not a "
+                           "genesis block. %s", block.identifier)
+
+    def _save_block_chain_id(self, block_chain_id):
```

Comment:
I don't like that the Journal/ChainController has to know the data_dir and now we have two separate locations in the code maintaining the block_chain_id_file. Where should the management of these data dir files be?